pyomo/solvers/plugins/solvers/scip_direct.py

```diff
@@ -0,0 +1,838 @@
+            )
+            self.results.solver.termination_condition = TerminationCondition.unknown
+            soln.status = SolutionStatus.stoppedByLimit
+        elif status == "nodelimit" or status == "totalnodelimit":
+            self.results.solver.status = SolverStatus.aborted
+            self.results.solver.termination_message = (
+                "Optimization terminated because the number of "
+                "branch-and-cut nodes explored exceeded the limits specified "
+                "in the limits/nodes or limits/totalnodes parameter"
+            )
+            self.results.solver.termination_condition = (
+                TerminationCondition.maxEvaluations
+            )
+            soln.status = SolutionStatus.stoppedByLimit
+        elif status == "timelimit":
+            self.results.solver.status = SolverStatus.aborted
+            self.results.solver.termination_message = (
+                "Optimization terminated because the time expended exceeded "
+                "the value specified in the limits/time parameter."
+            )
+            self.results.solver.termination_condition = (
+                TerminationCondition.maxTimeLimit
+            )
+            soln.status = SolutionStatus.stoppedByLimit
+        elif status == "sollimit" or status == "bestsollimit":
+            self.results.solver.status = SolverStatus.aborted
+            self.results.solver.termination_message = (
+                "Optimization terminated because the number of solutions found "
+                "reached the value specified in the limits/solutions or"
+                "limits/bestsol parameter."
+            )
+            self.results.solver.termination_condition = TerminationCondition.unknown
+            soln.status = SolutionStatus.stoppedByLimit
+        elif status == "memlimit":
+            self.results.solver.status = SolverStatus.aborted
+            self.results.solver.termination_message = (
+                "Optimization terminated because the memory used exceeded "
+                "the value specified in the limits/memory parameter."
+            )
+            self.results.solver.termination_condition = TerminationCondition.unknown
+            soln.status = SolutionStatus.stoppedByLimit
+        elif status == "userinterrupt":
+            self.results.solver.status = SolverStatus.aborted
+            self.results.solver.termination_message = (
+                "Optimization was terminated by the user."
+            )
+            self.results.solver.termination_condition = TerminationCondition.error
+            soln.status = SolutionStatus.error
+        else:
+            self.results.solver.status = SolverStatus.error
+            self.results.solver.termination_message = (
+                "Unhandled SCIP status (" + str(status) + ")"
+            )
+            self.results.solver.termination_condition = TerminationCondition.error
+            soln.status = SolutionStatus.error
```

Comment:
I recommend putting all of this in its own helper function. It'll clean up `_postsolve`.

Reply:
Done as suggested